sql/src/main/java/io/crate/planner/consumer/DistributedGroupByConsumer.java

```diff
@@ -82,26 +81,25 @@ public Visitor(Functions functions) {
 
         @Override
         public PlannedAnalyzedRelation visitQueriedDocTable(QueriedDocTable table, ConsumerContext context) {
-            QuerySpec querySpec = table.querySpec();
-            List<Symbol> groupBy = querySpec.groupBy();
-            if (groupBy == null) {
+            if (!table.querySpec().groupBy().isPresent()) {
                 return null;
             }
-
+            QuerySpec querySpec = table.querySpec();
+            List<Symbol> groupBy = querySpec.groupBy().get();
             DocTableInfo tableInfo = table.tableRelation().tableInfo();
             if(querySpec.where().hasVersions()){
                 context.validationException(new VersionInvalidException());
                 return null;
             }
 
-            GroupByConsumer.validateGroupBySymbols(table.tableRelation(), querySpec.groupBy());
+            GroupByConsumer.validateGroupBySymbols(table.tableRelation(), querySpec.groupBy().get());
```

Comment:
can you use the local groupBy here and further below instead of doing the repitive .get calls?